### PR TITLE
adding FileUpload injectable

### PR DESCRIPTION
**Pull request by benschw**

### CODE REVIEW

File: src/DefaultInjectablesFactory.php [hunk 1]
```diff
@@ -135,4 +137,20 @@ function(Context $context, $paramName) {
...
 		);
 	}
+
+	public function createFileUploadParam() {
+		return new InjectableProperty(
+			'Fliglio\Web\FileUpload',
+			function(Context $context, $paramName) {
+				$files = $context->getRequest()->getFiles();
+
+				if (!isset($files[$paramName])) {
```
Comment (oh4real) [inline src/DefaultInjectablesFactory.php:147]: Not entirely obvious how the method signature/argument $paramName correlates to an uploaded file input?
e.g. 
```
Resource::uploadedFile(FileUpload $anyRandoName)
```

must match:
```
<input type='file' name='anyRandoName' />
```

???

Comment (benschw) [reply src/DefaultInjectablesFactory.php:147]: thats exactly right, it follows the convention used by get params & path params. 

`$_FILES[$paramName]` targets a specific file upload

should I add something to make that more explicit?

Comment (benschw) [reply src/DefaultInjectablesFactory.php:147]: ...and you target it with the name of your injected param:
```
public function myResourceMethod(FileUpload $anyRandoName) { ... }
```

**Review by oh4real — APPROVED** [review]

**Comment by benschw** [comment]
FYI tests pass locally against the fliglio-web branch, I will rekick to confirm after web is merged and available in packagist

**Review by sven7777 — APPROVED** [review]

**Pull request merged**